### PR TITLE
Post new comment

### PR DESCRIPTION
**Pull request by liampjh34**
1. Add ability to post new comment with optimistic rendering
2. Add validation for empty comments

### CODE REVIEW

File: src/Components/CommentComposer.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        event.preventDefault()
+        setCommentLoading(true)
+        setSubmitAttempted(true)
+        if (!validateForm()) {
+            setCommentLoading(false)
+            return
+        }
+        
+        setComments([{
+            comment_id: 'temporaryKey',
```
Comment (MKD01) [inline src/Components/CommentComposer.jsx:42]: This isnt very unique so if a user creates two comments then they would both have this exact same comment_id which isnt ideal.

File: src/Components/CommentComposer.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            votes: 0
+        }, ...comments])
+        setComment('')
+        postComment(user, comment, articleId)
+        .then((response) => {
+            setCommentLoading(false)
+        })
+        .catch((error) => {
+            setError(error)
+            setCommentLoading(false)
```
Comment (MKD01) [inline src/Components/CommentComposer.jsx:54]: In the event the comment fails, we should remove the newly added comment from the comments list as well to show that is has not been added. Currently the newly added comment still stays in the list even when the request was unsuccessful.

**Review by MKD01 — COMMENTED** [review]
Really good work, but do check the comments I have left below.

File: src/Components/CommentComposer.jsx (hunk 2 above)
Comment (liampjh34) [reply src/Components/CommentComposer.jsx:54]: Thanks MKD! Fixed in https://github.com/liampjh34/nc-news/pull/9 ✅

File: src/Components/CommentComposer.jsx (hunk 1 above)
Comment (liampjh34) [reply src/Components/CommentComposer.jsx:42]: Thanks MKD! Fixed in https://github.com/liampjh34/nc-news/pull/9 ✅

**Pull request merged**